src/NuGet.Clients/NuGetConsole.Host.PowerShell/PowerShellHost.cs

```diff
@@ -325,6 +327,16 @@ public void Initialize(IConsole console)
 
                                 DefaultProject = null;
 
+                                var telemetryEvent = new TelemetryEvent("PowerShellExecuteCommand", new Dictionary<string, object>
```

Comment:
It might be helpful to add abstractions for the telemetry events, similar to RestoreCommand.

Reply:
When is being sent again?
It looks to me that it'll only be sent after init.ps1 runs.

Reply:
This one is hooked into `close solution event`. So it'll be only sent when user close the VS solution regardless of init.ps1.

Reply:
That abstraction is useful for wrapper around method and emit duration. But here only emit once for `solution close event` so I don't see any usefulness, having `using` statement with empty body doesn't make any sense.

Reply:
> It might be helpful to add abstractions for the telemetry events, similar to RestoreCommand.

That abstraction is useful for wrapper around method and emit duration. But here only emit once for `solution close event` so I don't see any usefulness, having `using` statement with empty body doesn't make any sense.

Reply:
> When is being sent again?
> It looks to me that it'll only be sent after init.ps1 runs.

This one is hooked into `close solution event`. So it'll be only sent when user close the VS solution regardless of init.ps1.